### PR TITLE
Refactor code with more uses of Map's Entry API

### PR DESCRIPTION
**Pull request by OBorce**
Small improvements.

### CODE REVIEW

File: accounting/src/delta/delta_amount_collection.rs [hunk 1]
```diff
@@ -58,28 +61,29 @@ impl<K: Ord> DeltaAmountCollection<K> {
...
+                // if the new amount is zero, no need to have it at all since it has no effect
+                if new_bal == SignedAmount::ZERO {
+                    drop(current.remove_entry());
+                } else {
+                    *current.get_mut() = new_bal;
+                }
+            }
+            Entry::Vacant(entry) => {
+                entry.insert(other_amount);
+            }
```
Comment (TheQuantumPhysicist) [inline accounting/src/delta/delta_amount_collection.rs:76]: This seems to me more unnecessarily verbose.

If you'd like to get rid of that `Option<SignedAmount>` in `combine_signed_amount_delta`, just change `current.get()` to `*current.unwrap_or(&SignedAmount::ZERO)`... not a huge benefit though.

**Review by TheQuantumPhysicist — APPROVED** [review]
Please revert that one change I commented on (unless we agree otherwise), then feel free to merge.

File: accounting/src/delta/delta_amount_collection.rs (hunk 1 above)
Comment (OBorce) [reply accounting/src/delta/delta_amount_collection.rs:76]: It was more about replacing the get + remove/insert on the map than the Option but ok.

**Review by muursh — APPROVED** [review]

File: accounting/src/delta/delta_amount_collection.rs (hunk 1 above)
Comment (OBorce) [reply accounting/src/delta/delta_amount_collection.rs]: reverted

**Pull request merged**